### PR TITLE
[ext.pages] Add `Paginator.edit()` method

### LINKED ISSUE #1093: Add a way to pass an interaction for a paginator to update
### Summary

Addition of a way to pass an existing interaction for a paginator to update

### What is the feature request for?

discord.ext.pages

### The Problem

Currently, I don't know of a clean and simple way to tell a paginator to update an existing interaction as opposed to sending the paginator as a new response, as in the following scenario:

```py
if condition:
  initial = await ctx.respond(embed=some_embed, view=some_view)
else:
  initial = None

if initial:
"""
Edit initial to be the paginator. So replace its embeds, message content, view, etc. as if it was a totally new response, but its editing the existing interaction
"""
else:
  await paginator.respond(ctx.interaction)
```

### The Ideal Solution

To be able to pass in an interaction for the paginator to edit

### The Current Solution

I was able to replicate the functionality I want by doing the following:
```py
if initial:
    paginator.user = ctx.author
    paginator.message = await initial.original_message()
    paginator.update_buttons()
    embed = paginator.pages[paginator.current_page]
    embed = paginator.get_page_content(embed)
    await initial.edit_original_message(view=paginator, embeds=embed)
else:
    await paginator.respond(ctx.interaction)
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by krittick**
## Summary

This adds a new `edit` method to the `Paginator` class, allowing users to more easily edit an existing message with the paginator contents.

Properly fixes #1093 

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

File: discord/ext/pages/pagination.py [hunk 1]
```diff
@@ -839,6 +839,73 @@ async def send(
...
+                embeds=page_content.embeds,
+                view=self,
+                suppress=suppress,
+                allowed_mentions=allowed_mentions,
+                delete_after=delete_after,
+            )
+        except discord.NotFound:
+            pass
+        except discord.Forbidden:
+            pass
```
Comment (Middledot) [inline discord/ext/pages/pagination.py:905]:
```suggestion
        except (discord.NotFound, discord.Forbidden):
            pass
```

File: discord/ext/pages/pagination.py [hunk 2]
```diff
@@ -839,6 +839,73 @@ async def send(
...
+
+        return message
```
Comment (Middledot) [inline discord/ext/pages/pagination.py:907]: Wouldn't this return the outdated message?

Comment (krittick) [reply discord/ext/pages/pagination.py]: > Wouldn't this return the outdated message?

Good catch, updated.

File: discord/ext/pages/pagination.py [hunk 3]
```diff
@@ -839,6 +839,70 @@ async def send(
...
+            to the object, otherwise it uses the attributes set in :attr:`~discord.Client.allowed_mentions`.
+            If no object is passed at all then the defaults given by :attr:`~discord.Client.allowed_mentions`
+            are used instead.
+        delete_after: Optional[:class:`float`]
+            If set, deletes the paginator after the specified time.
+
+        Returns
+        --------
+        :class:`discord.Message`
+            The message that was edited.
```
Comment (Middledot) [inline discord/ext/pages/pagination.py:878]:
```suggestion
        Optional[:class:`discord.Message`]
            The message that was edited. Returns ``None`` if the operation failed.
```

**Review by Middledot — APPROVED** [review]

**Pull request merged**